platforms/sdl/main.cpp

```diff
@@ -305,6 +316,10 @@ void CheckOptionalTextureAvailability()
 // Main
 int main(int argc, char *argv[])
 {
+	// Setup Logging
+	new Logger;
```

Comment:
> Does `Logger`'s constructor register that instance anywhere?

Yup.

Ideally the singleton would be initialized in `Logger.cpp`. The issue is the this would prevent the Win32 code from allocating its own `LoggerWin32`. So it starting the logger must be the responsibility of `main`. It's ugly, but it works without double-allocating or double-freeing anything.

Reply:
No but I mean, the constructor shouldn't be the one assigning the singleton, it looks weird and like a memory leak here. (Even though it's not)

Reply:
Fixed.